src/Mechanic/AstSymbolCollector.fs

```diff
@@ -0,0 +1,71 @@
+module Mechanic.AstSymbolCollector
+open Microsoft.FSharp.Compiler.Ast
+open Microsoft.FSharp.Compiler.SourceCodeServices.AstTraversal
+
+let visitLongIdent (ident: LongIdent) =
+    let names = String.concat "." [ for i in ident -> i.idText ]
+    names
+
+let visitPattern = function
+    | SynPat.Named(SynPat.Wild(_), name, _, _, _) -> Some name.idText
+    | SynPat.Named(_, name, _, _, _) -> Some name.idText
+    | SynPat.LongIdent(LongIdentWithDots(ident, _), _, _, _, _, _) -> Some <| visitLongIdent ident
```

Comment:
Is it worth creating some active patterns to hide these somewhat ugly repeated wildcards?

Reply:
Agreed.